### ISSUE TITLE
Is this implementation reference on this paper?

### ISSUE THREAD

**Issue by Da-Capo** [rarilurelo/XNOR-Net#1]
Is this implementation reference on this paper?
http://pjreddie.com/media/files/papers/xnor.pdf

**Comment by rarilurelo** [comment]
Yes, but now it's only implemented Binary Weighted Network.
XNOR-Net nearly close to BinaryNet(http://arxiv.org/abs/1602.02830), and I implemented it at rarilurelo/BinaryNetConvolution.
I think to make a little change at bst.py, it will work.

**Comment by Da-Capo** [comment]
thinks for your reply .
1. I have reivewd and your code and binarynet(https://github.com/MatthieuCourbariaux/BinaryNet) many times.I find you both give up to use the shift base batchnormal. is it becuease the effect is bad?or it's hard to implement in the framework like theano or chainer
2. what's the different between XNOR-Net and BNN? Is the XNOR-Net just changed the position of BN in the network?
![image](https://cloud.githubusercontent.com/assets/10624004/15246848/856447cc-1942-11e6-8820-4440ec8aaeb4.png)
![image](https://cloud.githubusercontent.com/assets/10624004/15246713/8e20cdc8-1941-11e6-8a24-b14c1d27836d.png)

**Comment by rarilurelo** [comment]
1. I didn't try to implement shift based batchnorm, because it's hard for me to implement backprop of shift based batchnorm. With theano or tensorflow, it will be easier.
2. I think the most different part between XNOR-Net and BNN is to use scaling factor or not.
   BNN's convolution is just convolved by 1 or -1, but XNOR-Net's convolution is multipled by scaling factor of weight after convolved by 1 or -1.

**Comment by Da-Capo** [comment]
thanks  i got it
